bitcoin/src/blockdata/fee_rate.rs

```diff
@@ -176,6 +173,80 @@ impl Div<Weight> for Amount {
     }
 }
 
+#[cfg(feature = "serde")]
+pub mod serde {
+    #![allow(missing_docs)]
+    //! A module for serde-serializing fee rates in various units.
+
+    use serde::de::{Deserialize, Deserializer, Error};
+    use serde::ser::Serializer;
+
+    use super::{Amount, FeeRate};
+    use crate::amount::Denomination;
+
+    pub mod btc_per_kvb {
+        //! Serialize fee rates as bitcoin per kilo-virtual bytes.
+        use super::*;
+
+        pub fn serialize<S: Serializer>(r: &FeeRate, s: S) -> Result<S::Ok, S::Error> {
+            s.serialize_f64((r.to_per_vb_ceil() * 1000).to_btc())
+        }
+
+        pub fn deserialize<'de, D: Deserializer<'de>>(d: D) -> Result<FeeRate, D::Error> {
+            let btc_per_kvb = f64::deserialize(d)?;
+            let per_kwu = Amount::from_btc(btc_per_kvb / 4.0).map_err(D::Error::custom)?;
+            Ok(FeeRate::from_per_kwu(per_kwu))
+        }
+
+        pub mod opt {
+            use super::*;
+
+            pub fn serialize<S: Serializer>(r: &Option<FeeRate>, s: S) -> Result<S::Ok, S::Error> {
+                match r {
+                    Some(r) => s.serialize_f64((r.to_per_vb_ceil() * 1000).to_btc()),
+                    None => s.serialize_none(),
+                }
+            }
+
+            pub fn deserialize<'de, D: Deserializer<'de>>(d: D) -> Result<Option<FeeRate>, D::Error> {
+                let btc_per_kvb = f64::deserialize(d)?;
+                let per_kwu = Amount::from_btc(btc_per_kvb / 4.0).map_err(D::Error::custom)?;
+                Ok(Some(FeeRate::from_per_kwu(per_kwu)))
+            }
+        }
+    }
+
+    pub mod sat_per_vb {
+        use super::*;
+
+        pub fn serialize<S: Serializer>(r: &FeeRate, s: S) -> Result<S::Ok, S::Error> {
+            s.serialize_f64(r.to_per_vb_ceil().to_float_in(Denomination::Satoshi))
+        }
+
+        pub fn deserialize<'de, D: Deserializer<'de>>(d: D) -> Result<FeeRate, D::Error> {
+            let sat_per_vb = Amount::from_float_in(f64::deserialize(d)?, Denomination::Satoshi)
+                .map_err(D::Error::custom)?;
+            Ok(FeeRate::from_per_vb(sat_per_vb).ok_or(D::Error::custom("overflow"))?)
+        }
+
+        pub mod opt {
+            use super::*;
+
+            pub fn serialize<S: Serializer>(r: &Option<FeeRate>, s: S) -> Result<S::Ok, S::Error> {
+                match r {
+                    Some(r) => s.serialize_f64(r.to_per_vb_ceil().to_float_in(Denomination::Satoshi)),
+                    None => s.serialize_none(),
+                }
+            }
+
+            pub fn deserialize<'de, D: Deserializer<'de>>(d: D) -> Result<Option<FeeRate>, D::Error> {
+                let sat_per_vb = Amount::from_float_in(f64::deserialize(d)?, Denomination::Satoshi)
+                    .map_err(D::Error::custom)?;
+                Ok(Some(FeeRate::from_per_vb(sat_per_vb).ok_or(D::Error::custom("overflow"))?))
+            }
+        }
+    }
```

Comment:
We also need `sat_per_kwu` which does come up in some protocols, is lossless, and was the original serialization.

Reply:
I'm curious which protocols use weight units in their feerate definitions. I don't think I've ever seen it. But I have been a bit inactive following protocol development lately. Does lightning to msat/wu? :)